CMakeLists.txt

```diff
@@ -1,18 +1,28 @@
-cmake_minimum_required(VERSION 3.16)
-# https://semver.org/
+cmake_minimum_required(VERSION 3.24)
```

Comment:
Can we avoid upgrading this? It's best practice to use the oldest version acceptable so building for older systems is still possible.

Reply:
I've been watching a lot of cmake talks and one of the recommendations most talks make is not to be afraid of requiring a very recent cmake version. Cmake is still changing quickly, there are a lot of valuable APIs in recent versions, and the cmake binary itself is easy to upgrade (doesn't require sudo, or system-wide install, etc). 

But acknowledged, 3.24 _might_ not be required here.